src/main/java/uk/gov/companieshouse/client/CompanyProfileClient.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.companieshouse.client;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.config.ApiConfig;
+import uk.gov.companieshouse.config.CompanyProfileConfig;
+
+@Service
+public class CompanyProfileClient {
+
+    private static class CompanyNotFoundException extends RuntimeException {}
+
+    private static final UriTemplate GET_COMPANY_URI = new UriTemplate("/company/{companyNumber}");
+
+    private static final String HEADER_AUTHORIZATION = "Authorization";
+    private static final String HEADER_ACCEPT = "Accept";
+    private static final String HEADER_CONTENT_TYPE = "Content-Type";
+
+    private static final String HEADER_ACCEPT_VALUE = "application/json";
+    private static final String HEADER_CONTENT_TYPE_VALUE = "application/json";
+
+    private final ApiConfig apiConfig;
+    private final CompanyProfileConfig companyProfileConfig;
+
+    @Autowired
+    public CompanyProfileClient(ApiConfig apiConfig, CompanyProfileConfig companyProfileConfig) {
```

Comment:
I think CompanyProfileConfig should extend ApiConfig, that way you only need to inject CompanyProfileConfig in here.